src/ProtocolZoo/ProtocolZoo.jl

```diff
@@ -212,16 +212,19 @@ end
     end
 end
 
-function findswapablequbits(net,node) # TODO parameterize the query predicates and the findmin/findmax
+function findswapablequbits(net, node, pred_low, pred_high) # TODO parameterize the query predicates and the findmin/findmax
     reg = net[node]
+    
+    l = typeof(pred_low) == Wildcard ? <(node) : pred_low
+    h = typeof(pred_high) == Wildcard ? >(node) : pred_high
 
-    leftnodes  = queryall(reg, EntanglementCounterpart, <(node), ❓; locked=false, assigned=true)
-    rightnodes = queryall(reg, EntanglementCounterpart, >(node), ❓; locked=false, assigned=true)
+    low_nodes  = queryall(reg, EntanglementCounterpart, l, ❓; locked=false, assigned=true)
+    high_nodes = queryall(reg, EntanglementCounterpart, h, ❓; locked=false, assigned=true)
 
-    (isempty(leftnodes) || isempty(rightnodes)) && return nothing
-    _, il = findmin(n->n.tag[2], leftnodes) # TODO make [2] into a nice named property
-    _, ir = findmax(n->n.tag[2], rightnodes)
-    return leftnodes[il], rightnodes[ir]
+    (isempty(low_nodes) || isempty(high_nodes)) && return nothing
+    _, il = findmin(n->n.tag[2], low_nodes) # TODO make [2] into a nice named property
+    _, ih = findmax(n->n.tag[2], high_nodes)
+    return low_nodes[il], high_nodes[ih]
 end
```

Comment:
Is the change in naming scheme seem suitable, I was hoping to come up with something more general since we are considering direction in a 2d grid as well now, instead of just left and right

Reply:
Yup, that is a good idea! Thanks!